nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,428 @@
+            Mask to apply to regions before extracting signals. Defines the \
+            overall area of the brain to consider. The data for each \
+            hemisphere is of shape (n_vertices/2, n_regions).
+
+        allow_overlap : :obj:`bool`, default=True
+            If False, an error is raised if the maps overlaps (ie at least two
+            maps have a non-zero value for the same voxel).
+
+        %(smoothing_fwhm)s
+            This parameter is not implemented yet.
+
+        %(standardize_maskers)s
+
+        %(standardize_confounds)s
+
+        %(detrend)s
+
+        high_variance_confounds : :obj:`bool`, default=False
+            If True, high variance confounds are computed on provided image \
+            with :func:`nilearn.image.high_variance_confounds` and default \
+            parameters and regressed out.
+
+        %(low_pass)s
+
+        %(high_pass)s
+
+        %(t_r)s
+
+        %(memory)s
+
+        %(memory_level1)s
+
+        %(verbose0)s
+
+        reports : :obj:`bool`, default=True
+            If set to True, data is saved in order to produce a report.
+
+        %(cmap)s
+            default="inferno"
+            Only relevant for the report figures.
+
+        clean_args : :obj:`dict` or None, default=None
+            Keyword arguments to be passed
+            to :func:`nilearn.signal.clean`
+            called within the masker.
+
+    Attributes
+    ----------
+        maps_img_ : :obj:`~nilearn.surface.SurfaceImage`
+            The same as the input `maps_img`, kept solely for consistency
+            across maskers.
+
+        n_elements_ : :obj:`int`
+            The number of regions in the maps image.
+
+    See Also
+    --------
+        nilearn.maskers.SurfaceMasker
+        nilearn.maskers.SurfaceLabelsMasker
+
+    """
+
+    def __init__(
+        self,
+        maps_img=None,
+        mask_img=None,
+        allow_overlap=True,
+        smoothing_fwhm=None,
+        standardize=False,
+        standardize_confounds=True,
+        detrend=False,
+        high_variance_confounds=False,
+        low_pass=None,
+        high_pass=None,
+        t_r=None,
+        memory=None,
+        memory_level=1,
+        verbose=0,
+        reports=True,
+        cmap="inferno",
+        clean_args=None,
+    ):
+        self.maps_img = maps_img
+        self.mask_img = mask_img
+        self.allow_overlap = allow_overlap
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.standardize_confounds = standardize_confounds
+        self.high_variance_confounds = high_variance_confounds
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.reports = reports
+        self.cmap = cmap
+        self.clean_args = clean_args
```

Comment:
do you think you can add a smoke test that uses the cleaning part of the masker: it seems that we are not testing this systematically for our maskers.

Reply:
Done!

But it seems that the way we expect arguments for clean is not consistent between Nifti and Surface maskers as of now:

NiftiMasker:
<img width="832" alt="Screenshot 2024-12-09 at 5 21 34 PM" src="https://github.com/user-attachments/assets/43326dc8-620b-4c63-a6ce-5051ff142259">

SurfaceMasker:
<img width="759" alt="Screenshot 2024-12-09 at 5 22 12 PM" src="https://github.com/user-attachments/assets/f850c382-975d-4690-a52a-5b04cf63dd5f">

Reply:
* I think there is a related issue #4805